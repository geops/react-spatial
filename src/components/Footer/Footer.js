import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  /**
   * CSS class of the button.
   */
  className: PropTypes.string,

  /**
   * Element to display in a container.
   */
  children: PropTypes.node,
};

const defaultProps = {
  children: undefined,
  className: 'tm-footer',
};

/**
 * This component displays a div which stick to the bottom of a container.
 */
class Footer extends PureComponent {
  render() {
    const { children, className } = this.props;

    return <div className={className}>{children}</div>;
  }
}

Footer.propTypes = propTypes;
Footer.defaultProps = defaultProps;

export default Footer;
