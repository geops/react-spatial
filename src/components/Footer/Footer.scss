.tm-footer {
  position: absolute;
  bottom: 0;
  right: 0;
  left: 0;
  text-align: right;
  font-size: 0.8em;
  padding: 5px;
  border-top: 1px solid #aaa;
  background: white;

  @include a();

  // To be removed
  a {
    margin-left: 10px;
    color: black;
  }
}
