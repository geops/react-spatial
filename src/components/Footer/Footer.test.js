import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Footer from './Footer';

configure({ adapter: new Adapter() });

describe('Footer', () => {
  test('matches snapshot', () => {
    const right = "I'm the left component";
    const component = renderer.create(<Footer right={right} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
