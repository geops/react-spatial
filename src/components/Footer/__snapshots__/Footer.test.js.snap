// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `
<div
  className="tm-footer"
/>
`;
